File: SNS.Web/src/app/support/support-tickets/contracts/support-ticket-details.dto.ts
import { SupportTicketCategory } from '../../enums/support-ticket-category.enum';
import { TicketPriority } from '../../enums/ticket-priority.enum';
import { TicketStatus } from '../../enums/ticket-status.enum';
import { TicketMessageDto } from '../../ticket-messages/contracts/ticket-message.dto';

export interface SupportTicketDetailsDto {
    id: string;
    userId: string;
    assignedAgentId: string | null;
    title: string;
    category: SupportTicketCategory;
    priority: TicketPriority;
    status: TicketStatus;
    createdAt: string;
    updatedAt: string;
    messages: TicketMessageDto[];
}

File: backend/SNS.Application/Support/SupportTickets/Contracts/SupportTicketDetailsDto.cs
using SNS.Application.Support.TicketMessages.Contracts;
using SNS.Domain.Support.Enums;

namespace SNS.Application.Support.SupportTickets.Contracts;

public sealed record SupportTicketDetailsDto(
    Guid Id,
    Guid UserId,
    Guid? AssignedAgentId,
    string Title,
    SupportTeckitCategory Category,
    TicketPriority Priority,
    TicketStatus Status,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    IReadOnlyList<TicketMessageDto> Messages);
